src/clj/puppetlabs/services/jruby_pool_manager/impl/jruby_internal.clj

```diff
@@ -42,18 +46,38 @@
     :force RubyInstanceConfig$CompileMode/FORCE
     :off RubyInstanceConfig$CompileMode/OFF))
 
+(schema/defn ^:always-validate get-profiling-mode :- RubyInstanceConfig$ProfilingMode
+  [config-profiling-mode :- jruby-schemas/SupportedJRubyProfilingModes]
+  (case config-profiling-mode
```

Comment:
👍  Matches http://jruby.org/apidocs/org/jruby/RubyInstanceConfig.ProfilingMode.html

I know no Clojure - but... any way to automatically map the available constants in `RubyInstanceConfig` to keywords (I think that's what `:api` is?) rather than hard-coding the mapping?

Reply:
We could automate that mapping assuming theres' a way to ask a Java class what its constants are (I don't know how to do that off the top of my head) and then stringify them and parse the strings. But I think that's a little more gnarly than just hardcoding them?

Reply:
Maybe. I might be able to grab the enum values and map them to keywords.

This seems to work:
```
(map keyword (map (comp clojure.string/lower-case str) (RubyInstanceConfig$ProfilingMode/values)))
(:off :api :flat :graph :html :json :service)
```

Reply:
`(map (comp keyword clojure.string/lower-case str) (RubyInstanceConfig$ProfilingMode/values))`

Reply:
I think then to grab the right constant from a given keyword we would do something like.

```(RubyInstanceConfig$ProfilingMode/valueOf (clojure.string/upper-case (name :off)))
=> #object[org.jruby.RubyInstanceConfig$ProfilingMode 0x4d53e068 "OFF"]
```

Reply:
Oh nice, that worked great

Reply:
(I was following the previous pattern used for compile-mode here)

Reply:
The hardcoding is sorta useful because people can look into this code to see what values they can pick from

Reply:
I personally prefer hardcoding it for that reason, but I agree it's more brittle.

Reply:
I think that's what the schema / `supported-jruby-profiling-modes` is valuable for.

Reply:
I'm going to leave the list for the schema, and update this section to remove the case